votor/src/event_handler/stats.rs

```diff
@@ -0,0 +1,312 @@
+use {
+    crate::{common::VoteType, event::VotorEvent, voting_service::BLSOp},
+    agave_votor_messages::consensus_message::ConsensusMessage,
+    solana_clock::Slot,
+    solana_metrics::datapoint_info,
+    std::{
+        collections::{BTreeMap, HashMap},
+        time::{Duration, Instant},
+    },
+};
+
+const STATS_REPORT_INTERVAL: Duration = Duration::from_secs(1);
+
+#[derive(Debug, Clone)]
+struct SlotTracking {
+    /// The time when the slot tracking started
+    start: Instant,
+    /// The time when the first shred for this slot was received
+    first_shred: Option<Instant>,
+    /// The time when the parent block for this slot was ready
+    parent_ready: Option<Instant>,
+    /// The time when the notarization vote for this slot was sent
+    vote_notarize: Option<Instant>,
+    /// The time when the skip vote for this slot was sent
+    vote_skip: Option<Instant>,
+    /// If the slot was finalized, this is the time when it was finalized,
+    /// the bool indicates if it was fast finalized
+    finalized: Option<(Instant, bool)>,
+}
+
+impl Default for SlotTracking {
+    fn default() -> Self {
+        Self {
+            start: Instant::now(),
+            first_shred: None,
+            parent_ready: None,
+            vote_notarize: None,
+            vote_skip: None,
+            finalized: None,
```

Comment:
nit:
start: Instant::now(),
..Default::default()

Reply:
throws the self reference recursion bug. But I can have the individual fields set to Default